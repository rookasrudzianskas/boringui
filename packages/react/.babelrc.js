module.exports = (api) => {
  const env = api.env();

  const removePropsPlugin = [
    "react-remove-properties",
    {
      properties: ["data-testid"],
    },
  ];

  let dev = false;
  let modules;
  let plugins;

  switch (env) {
    case "docs":
    case "test":
    case "dist-dev":
    case "development":
      dev = true;
      modules = false;
      break;
    case "dist-prod":
      plugins = [removePropsPlugin];
    case "production":
      plugins = [removePropsPlugin];
    case "esm":
      modules = false;
      break;
    case "cjs":
    default:
      modules = "commonjs";
  }

  const presets =
    env !== "test" && env !== "development"
      ? [
          [
            "@react-bootstrap",
            {
              dev,
              modules,
              removePropTypes: !dev,
            },
          ],
          "@babel/preset-typescript",
        ]
      : ["@babel/preset-env", "@babel/react", "@babel/preset-typescript"];

  return {
    presets,
    plugins,
    ignore:
      env !== "test" && env !== "development"
        ? [/@babel[\\|/]runtime/, /\.stories\.(js|ts|tsx)$/, /\.test\.(js|ts|tsx)$/]
        : [],
  };
};
