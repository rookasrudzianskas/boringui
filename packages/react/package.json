{
  "name": "@nextui-org/react",
  "version": "1.0.0-beta.10",
  "license": "MIT",
  "description": "ðŸš€ Beautiful and modern React UI library.",
  "homepage": "https://nextui.org",
  "author": {
    "name": "Junior Garcia",
    "email": "jrgarciadev@gmail.com"
  },
  "bugs": {
    "url": "https://github.com/nextui-org/nextui/issues/new/choose"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/nextui-org/nextui"
  },
  "keywords": [
    "next",
    "next ui",
    "components",
    "modern components",
    "react components",
    "react ui"
  ],
  "files": [
    "lib",
    "dist",
    "es"
  ],
  "source": "src/index.ts",
  "main": "lib/cjs/index.js",
  "module": "lib/esm/index.js",
  "unpkg": "lib/umd/nextui.min.js",
  "types": "lib/types/index.d.ts",
  "sideEffects": false,
  "scripts": {
    "clear": "rimraf ./lib ./dist ./esm",
    "clear-types": "rimraf ./lib/types",
    "pre-publish": "node ./scripts/pre-publish.js",
    "publish:dry-run": "npm publish ./lib --dry-run",
    "build": "node ./scripts/build.js",
    "build:minify": "node ./scripts/minify.js",
    "build:types": "yarn tsc -p ./buildconfig -d --emitDeclarationOnly --outDir types",
    "build:dev-types": "tsc -p ./buildconfig -d --emitDeclarationOnly --outDir lib/esm",
    "build:webpack": "webpack --config ./buildconfig/webpack.common.js",
    "build:watch": "node ./scripts/build-watch.js",
    "dev": "yarn clear && yarn build:dev-types && yarn build:watch",
    "lint": "eslint \"src/**/*.{js,ts,tsx}\"",
    "test": "jest --config jest.config.js --no-cache",
    "test-update": "jest --config jest.config.js --no-cache --update-snapshot",
    "coverage": "yarn test --coverage",
    "coverage-update": "yarn test-update --coverage",
    "coverage:codecov": "yarn test --coverage && codecov"
  },
  "dependencies": {
    "@babel/runtime": "^7.6.2",
    "@react-aria/button": "3.6.0",
    "@react-aria/checkbox": "3.5.0",
    "@react-aria/dialog": "3.3.0",
    "@react-aria/focus": "3.7.0",
    "@react-aria/i18n": "3.5.1",
    "@react-aria/interactions": "3.10.0",
    "@react-aria/label": "3.4.0",
    "@react-aria/link": "3.3.1",
    "@react-aria/menu": "3.6.0",
    "@react-aria/overlays": "3.10.0",
    "@react-aria/radio": "3.3.0",
    "@react-aria/ssr": "3.3.0",
    "@react-aria/table": "3.4.0",
    "@react-aria/utils": "3.13.2",
    "@react-aria/visually-hidden": "3.4.0",
    "@react-stately/checkbox": "3.2.0",
    "@react-stately/data": "3.6.0",
    "@react-stately/overlays": "3.4.0",
    "@react-stately/radio": "3.5.0",
    "@react-stately/table": "3.3.0",
    "@react-stately/toggle": "3.4.0",
    "@react-stately/tree": "3.3.2",
    "@react-types/button": "^3.6.0",
    "@react-types/checkbox": "3.3.2",
    "@react-types/grid": "3.1.2",
    "@react-types/menu": "3.7.0",
    "@react-types/overlays": "3.6.2",
    "@react-types/shared": "3.14.0",
    "@stitches/react": "1.2.8"
  },
  "peerDependencies": {
    "react": ">=16.8.0",
    "react-dom": ">=16.8.0"
  },
  "publishConfig": {
    "directory": "lib"
  },
  "devDependencies": {
    "react": "17.0.2",
    "@babel/cli": "^7.14.5",
    "@babel/plugin-proposal-object-rest-spread": "^7.15.6",
    "@babel/plugin-transform-runtime": "^7.14.5",
    "@babel/preset-env": "^7.14.5",
    "@babel/preset-react": "^7.14.5",
    "@babel/preset-typescript": "^7.14.5",
    "@react-bootstrap/babel-preset": "^2.1.0",
    "@storybook/react": "^6.5.3",
    "@testing-library/dom": "^8.1.0",
    "@testing-library/react": "^12.1.2",
    "@testing-library/react-hooks": "^7.0.0",
    "@testing-library/user-event": "^13.2.0",
    "@types/enzyme": "^3.10.8",
    "@types/jest": "^26.0.22",
    "@types/node": "^15.12.4",
    "@types/react": "17.0.33",
    "@types/react-dom": "17.0.9",
    "@typescript-eslint/eslint-plugin": "^4.27.0",
    "@typescript-eslint/parser": "^4.27.0",
    "@wojtekmaj/enzyme-adapter-react-17": "^0.6.1",
    "babel-jest": "^26.6.3",
    "babel-loader": "^8.2.2",
    "babel-plugin-module-resolver": "^4.1.0",
    "babel-plugin-optimize-clsx": "^2.6.2",
    "babel-plugin-react-remove-properties": "^0.3.0",
    "babel-plugin-syntax-dynamic-import": "^6.18.0",
    "chalk": "^4.1.2",
    "cherry-pick": "^0.5.0",
    "css-mediaquery": "^0.1.2",
    "enzyme": "^3.11.0",
    "eslint-plugin-react": "^7.24.0",
    "execa": "^5.1.1",
    "fs-extra": "^10.0.0",
    "jest": "^26.6.3",
    "p-iteration": "^1.1.8",
    "parcel": "^2.3.1",
    "prettier": "^2.3.1",
    "rimraf": "^3.0.2",
    "terser": "5.10.0",
    "ts-jest": "^26.5.5",
    "typescript": "4.6.2",
    "webpack": "^5.53.0",
    "webpack-bundle-analyzer": "^4.4.2",
    "webpack-cli": "^3.3.11",
    "webpack-merge": "^5.8.0"
  }
}
