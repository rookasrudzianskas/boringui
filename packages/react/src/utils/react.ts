import * as React from "react";

export function getValidChildren(children: React.ReactNode) {
  return React.Children.toArray(children).filter((child) =>
    React.isValidElement(child),
  ) as React.ReactElement[];
}

export function addChildrenClass(children: React.ReactNode, className: string) {
  const validChildren = getValidChildren(children);

  return validChildren.map((child) => {
    return React.cloneElement(child, {
      className: `${child.props.className} ${className}`,
    });
  });
}
