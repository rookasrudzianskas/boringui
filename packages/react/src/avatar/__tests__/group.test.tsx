import React from "react";
import {mount, shallow} from "enzyme";
import {render} from "@testing-library/react";

import Avatar from "../index";

describe("AvatarGroup", () => {
  it("should render correctly", () => {
    const pictureUsers = [
      "https://i.pravatar.cc/300?u=a042581f4e29026705d",
      "https://i.pravatar.cc/300?u=a042581f4e29026706d",
      "https://i.pravatar.cc/300?u=a042581f4e29026707d",
      "https://i.pravatar.cc/300?u=a042581f4e29026709d",
      "https://i.pravatar.cc/300?u=a042581f4f29026709d",
    ];
    const nameUsers = ["Junior", "Jane", "W", "John"];
    const wrapper = mount(
      <Avatar.Group animated>
        {pictureUsers.map((url, index) => (
          <Avatar key={index} bordered stacked src={url} />
        ))}
      </Avatar.Group>,
    );

    expect(() => wrapper.unmount()).not.toThrow();
    const rendered = render(
      <Avatar.Group animated={false}>
        {nameUsers.map((name, index) => (
          <Avatar key={index} bordered stacked text={name} />
        ))}
      </Avatar.Group>,
    );

    expect(rendered).toMatchSnapshot();
  });

  it("group component should render all children", () => {
    const group = render(
      <Avatar.Group>
        <Avatar data-testid="avatar-test" text="1" />
        <Avatar data-testid="avatar-test" text="2" />
      </Avatar.Group>,
    );

    const avatars = group.getAllByTestId("avatar-test");

    expect(avatars).toHaveLength(2);
  });

  it("should stacked when avatars are in a group", () => {
    const group = render(
      <Avatar.Group>
        <Avatar text="1" />
        <Avatar text="2" />
      </Avatar.Group>,
    );

    expect(group).toMatchSnapshot();
  });

  it("should show count in group", () => {
    const count = 20;
    const group = shallow(<Avatar.Group count={count} />);
    const text = group.find(".nextui-avatar-group-count").text();

    expect(text).toContain(count);
  });
});
