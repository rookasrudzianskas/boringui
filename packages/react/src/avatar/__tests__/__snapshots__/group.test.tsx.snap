// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarGroup should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="nextui-c-cpgyzW"
      >
        <span
          class="nextui-c-gEEnVX nextui-c-PJLV nextui-c-gEEnVX-qXFXw-color-default nextui-c-gEEnVX-duCSmO-textColor-default nextui-c-gEEnVX-iHugVF-size-md nextui-c-gEEnVX-krOtXB-borderWeight-normal nextui-c-gEEnVX-dbZNQr-bordered-true nextui-c-gEEnVX-gBVgtO-stacked-true nextui-c-gEEnVX-bAEoQm-rounded-true nextui-c-gEEnVX-MtAEf-cv nextui-c-PJLV-gulvcB-isFocusVisible-false only-text-avatar"
          data-state="ready"
        >
          <span
            class="nextui-avatar-bg"
          />
          <span
            class="nextui-avatar-text"
          >
            Jun
          </span>
        </span>
        <span
          class="nextui-c-gEEnVX nextui-c-PJLV nextui-c-gEEnVX-qXFXw-color-default nextui-c-gEEnVX-duCSmO-textColor-default nextui-c-gEEnVX-iHugVF-size-md nextui-c-gEEnVX-krOtXB-borderWeight-normal nextui-c-gEEnVX-dbZNQr-bordered-true nextui-c-gEEnVX-gBVgtO-stacked-true nextui-c-gEEnVX-bAEoQm-rounded-true nextui-c-gEEnVX-MtAEf-cv nextui-c-PJLV-gulvcB-isFocusVisible-false only-text-avatar"
          data-state="ready"
        >
          <span
            class="nextui-avatar-bg"
          />
          <span
            class="nextui-avatar-text"
          >
            Jane
          </span>
        </span>
        <span
          class="nextui-c-gEEnVX nextui-c-PJLV nextui-c-gEEnVX-qXFXw-color-default nextui-c-gEEnVX-duCSmO-textColor-default nextui-c-gEEnVX-iHugVF-size-md nextui-c-gEEnVX-krOtXB-borderWeight-normal nextui-c-gEEnVX-dbZNQr-bordered-true nextui-c-gEEnVX-gBVgtO-stacked-true nextui-c-gEEnVX-bAEoQm-rounded-true nextui-c-gEEnVX-MtAEf-cv nextui-c-PJLV-gulvcB-isFocusVisible-false only-text-avatar"
          data-state="ready"
        >
          <span
            class="nextui-avatar-bg"
          />
          <span
            class="nextui-avatar-text"
          >
            W
          </span>
        </span>
        <span
          class="nextui-c-gEEnVX nextui-c-PJLV nextui-c-gEEnVX-qXFXw-color-default nextui-c-gEEnVX-duCSmO-textColor-default nextui-c-gEEnVX-iHugVF-size-md nextui-c-gEEnVX-krOtXB-borderWeight-normal nextui-c-gEEnVX-dbZNQr-bordered-true nextui-c-gEEnVX-gBVgtO-stacked-true nextui-c-gEEnVX-bAEoQm-rounded-true nextui-c-gEEnVX-MtAEf-cv nextui-c-PJLV-gulvcB-isFocusVisible-false only-text-avatar"
          data-state="ready"
        >
          <span
            class="nextui-avatar-bg"
          />
          <span
            class="nextui-avatar-text"
          >
            John
          </span>
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="nextui-c-cpgyzW"
    >
      <span
        class="nextui-c-gEEnVX nextui-c-PJLV nextui-c-gEEnVX-qXFXw-color-default nextui-c-gEEnVX-duCSmO-textColor-default nextui-c-gEEnVX-iHugVF-size-md nextui-c-gEEnVX-krOtXB-borderWeight-normal nextui-c-gEEnVX-dbZNQr-bordered-true nextui-c-gEEnVX-gBVgtO-stacked-true nextui-c-gEEnVX-bAEoQm-rounded-true nextui-c-gEEnVX-MtAEf-cv nextui-c-PJLV-gulvcB-isFocusVisible-false only-text-avatar"
        data-state="ready"
      >
        <span
          class="nextui-avatar-bg"
        />
        <span
          class="nextui-avatar-text"
        >
          Jun
        </span>
      </span>
      <span
        class="nextui-c-gEEnVX nextui-c-PJLV nextui-c-gEEnVX-qXFXw-color-default nextui-c-gEEnVX-duCSmO-textColor-default nextui-c-gEEnVX-iHugVF-size-md nextui-c-gEEnVX-krOtXB-borderWeight-normal nextui-c-gEEnVX-dbZNQr-bordered-true nextui-c-gEEnVX-gBVgtO-stacked-true nextui-c-gEEnVX-bAEoQm-rounded-true nextui-c-gEEnVX-MtAEf-cv nextui-c-PJLV-gulvcB-isFocusVisible-false only-text-avatar"
        data-state="ready"
      >
        <span
          class="nextui-avatar-bg"
        />
        <span
          class="nextui-avatar-text"
        >
          Jane
        </span>
      </span>
      <span
        class="nextui-c-gEEnVX nextui-c-PJLV nextui-c-gEEnVX-qXFXw-color-default nextui-c-gEEnVX-duCSmO-textColor-default nextui-c-gEEnVX-iHugVF-size-md nextui-c-gEEnVX-krOtXB-borderWeight-normal nextui-c-gEEnVX-dbZNQr-bordered-true nextui-c-gEEnVX-gBVgtO-stacked-true nextui-c-gEEnVX-bAEoQm-rounded-true nextui-c-gEEnVX-MtAEf-cv nextui-c-PJLV-gulvcB-isFocusVisible-false only-text-avatar"
        data-state="ready"
      >
        <span
          class="nextui-avatar-bg"
        />
        <span
          class="nextui-avatar-text"
        >
          W
        </span>
      </span>
      <span
        class="nextui-c-gEEnVX nextui-c-PJLV nextui-c-gEEnVX-qXFXw-color-default nextui-c-gEEnVX-duCSmO-textColor-default nextui-c-gEEnVX-iHugVF-size-md nextui-c-gEEnVX-krOtXB-borderWeight-normal nextui-c-gEEnVX-dbZNQr-bordered-true nextui-c-gEEnVX-gBVgtO-stacked-true nextui-c-gEEnVX-bAEoQm-rounded-true nextui-c-gEEnVX-MtAEf-cv nextui-c-PJLV-gulvcB-isFocusVisible-false only-text-avatar"
        data-state="ready"
      >
        <span
          class="nextui-avatar-bg"
        />
        <span
          class="nextui-avatar-text"
        >
          John
        </span>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AvatarGroup should stacked when avatars are in a group 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="nextui-c-cpgyzW nextui-c-cpgyzW-jErlEr-animated-true"
      >
        <span
          class="nextui-c-gEEnVX nextui-c-PJLV nextui-c-gEEnVX-qXFXw-color-default nextui-c-gEEnVX-duCSmO-textColor-default nextui-c-gEEnVX-iHugVF-size-md nextui-c-gEEnVX-krOtXB-borderWeight-normal nextui-c-gEEnVX-bAEoQm-rounded-true nextui-c-PJLV-gulvcB-isFocusVisible-false only-text-avatar"
          data-state="ready"
        >
          <span
            class="nextui-avatar-bg"
          />
          <span
            class="nextui-avatar-text"
          >
            1
          </span>
        </span>
        <span
          class="nextui-c-gEEnVX nextui-c-PJLV nextui-c-gEEnVX-qXFXw-color-default nextui-c-gEEnVX-duCSmO-textColor-default nextui-c-gEEnVX-iHugVF-size-md nextui-c-gEEnVX-krOtXB-borderWeight-normal nextui-c-gEEnVX-bAEoQm-rounded-true nextui-c-PJLV-gulvcB-isFocusVisible-false only-text-avatar"
          data-state="ready"
        >
          <span
            class="nextui-avatar-bg"
          />
          <span
            class="nextui-avatar-text"
          >
            2
          </span>
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="nextui-c-cpgyzW nextui-c-cpgyzW-jErlEr-animated-true"
    >
      <span
        class="nextui-c-gEEnVX nextui-c-PJLV nextui-c-gEEnVX-qXFXw-color-default nextui-c-gEEnVX-duCSmO-textColor-default nextui-c-gEEnVX-iHugVF-size-md nextui-c-gEEnVX-krOtXB-borderWeight-normal nextui-c-gEEnVX-bAEoQm-rounded-true nextui-c-PJLV-gulvcB-isFocusVisible-false only-text-avatar"
        data-state="ready"
      >
        <span
          class="nextui-avatar-bg"
        />
        <span
          class="nextui-avatar-text"
        >
          1
        </span>
      </span>
      <span
        class="nextui-c-gEEnVX nextui-c-PJLV nextui-c-gEEnVX-qXFXw-color-default nextui-c-gEEnVX-duCSmO-textColor-default nextui-c-gEEnVX-iHugVF-size-md nextui-c-gEEnVX-krOtXB-borderWeight-normal nextui-c-gEEnVX-bAEoQm-rounded-true nextui-c-PJLV-gulvcB-isFocusVisible-false only-text-avatar"
        data-state="ready"
      >
        <span
          class="nextui-avatar-bg"
        />
        <span
          class="nextui-avatar-text"
        >
          2
        </span>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
